### PR TITLE
fix: qwen fc error

### LINKED ISSUE #6567: Some qwen llm can not invoke function-call
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.14

### Cloud or Self Hosted

Self Hosted (Source)

### Steps to reproduce

on 2024.07.23，I can not invoke fc with some qwen llm，like qwen-max... But actually,it work last week.
6fc774a5c3857d1cd156496fc140f6f

### ✔️ Expected Behavior

I hope it invoke fc correctly
d5d4eed8762fc1bd2d1f7ab531f2338

### ❌ Actual Behavior

However, qwen-max can not invoke correctly

### PR DESCRIPTION
**Pull request by Dofine-dufei**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description
Fixes #6567 

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api/core/model_runtime/model_providers/tongyi/llm/llm.py [hunk 1]
```diff
@@ -512,6 +515,19 @@ def _convert_prompt_messages_to_tongyi_messages(self, prompt_messages: list[Prom
 
         return tongyi_messages
 
+    def object_to_dict(self, obj):
```
Comment (laipz8200) [inline api/core/model_runtime/model_providers/tongyi/llm/llm.py]: This `obj` should be a `ToolCall` instance. Here is its structure: [link](https://github.com/langgenius/dify/blob/5a99aeb864092d2e3b31e5547ce827ab290e4447/api/core/model_runtime/entities/message_entities.py#L111).

It's better to add a type hint and avoid using the magic methods like `__dict__`.

**Review by laipz8200 — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/tongyi/llm/llm.py (hunk 1 above)
Comment (Dofine-dufei) [reply api/core/model_runtime/model_providers/tongyi/llm/llm.py]: ToolCall has parameter ToolCallFunction, it also need to transfer to dict.So I think my way is better.

Comment (Dofine-dufei) [reply api/core/model_runtime/model_providers/tongyi/llm/llm.py]: I referred to the call of alibaba qwen‘s way

Comment (laipz8200) [reply api/core/model_runtime/model_providers/tongyi/llm/llm.py]: How about using the `model_dump` method provided by Pydantic?

Comment (Dofine-dufei) [reply api/core/model_runtime/model_providers/tongyi/llm/llm.py]: Fine,I have tested just now. It is the best way with **model_dump**

**Review by laipz8200 — APPROVED** [review]
Great! Thank you for your contribution.

**Pull request merged**